lib/fprotect/fprotect_acl.c

```diff
@@ -56,6 +57,41 @@ static int fprotect_set_permission(uint32_t start, size_t length,
 	return 0;
 }
 
+#ifdef CONFIG_FPROTECT_ENABLE_IS_PROTECTED
+uint32_t fprotect_is_protected(uint32_t addr)
+{
+	uint32_t region_idx;
+	uint32_t page_size = nrf_ficr_codepagesize_get(NRF_FICR);
+	uint32_t flash_size = nrf_ficr_codesize_get(NRF_FICR) * page_size;
+	uint32_t block_addr = ROUND_DOWN(addr, CONFIG_FPROTECT_BLOCK_SIZE);
+
+	find_free_region(&region_idx);
+
+	for (int i = 0; i < region_idx; i++) {
+		uint32_t prot_addr = nrf_acl_region_address_get(NRF_ACL, i);
+		uint32_t prot_size = nrf_acl_region_size_get(NRF_ACL, i);
+		nrf_acl_perm_t prot_perm = nrf_acl_region_perm_get(NRF_ACL, i);
+
+		if ((prot_addr > flash_size) || (prot_size > NRF_ACL_REGION_SIZE_MAX)) {
+			/* ACL is incorrectly configured! */
+			k_panic();
+		}
+
+		uint32_t prot_end_addr = prot_addr + prot_size;
+		uint32_t end_addr = block_addr + CONFIG_FPROTECT_BLOCK_SIZE;
+
+		if ((prot_addr <= block_addr) && (prot_end_addr >= end_addr)) {
```

Comment:
I am probably missing something but why the protected address should be <= to the block_addr and not >=?

Reply:
This is the successful case, where the protection extends at least from `block_addr` to `end_addr`

Reply:
Alright, thanks!